### ISSUE TITLE
Random Refresh while reading articles

### ISSUE THREAD

**Issue by mediaformat** [wallabag/ios-app#417]
While reading longer articles, the content is replaced by a spinner, which never resolves.

Leaving the article and returning, the scroll position is lost.

iOS 17.4.1
Wallabag ios 7.2.3 build 319

**Comment by bourvill** [comment]
Hello, can you share the original link of your entry?

**Comment by mediaformat** [comment]
Sure, it happened with both of these:
- https://www.noemamag.com/we-need-to-rewild-the-internet/
- https://werd.io/2024/where-im-coming-from

**Comment by bourvill** [comment]
After multiple try to reproduce. it's never happened on my device. Can you record an example ?

**Comment by steveh** [comment]
I get a similar issue, the content is replaced by a white screen, and when loading again the scroll position is lost.

It happens on any long article that I’m reading normally and will typically first happen about 1/3rd through. It seems to become more frequent near the end but I’m not certain about that

**Comment by bourvill** [comment]
hi @steveh what is your iOS version and device ?

**Comment by steveh** [comment]
iPhone 14 Pro, v17.4.1. Wallabag v7.2.4 b319.
I just realised there is a newer iOS version so I'll try upgrade

**Comment by mediaformat** [comment]
> After multiple try to reproduce. it's never happened on my device. Can you record an example ?

https://github.com/wallabag/ios-app/assets/2895351/a0a1c356-7954-4bcc-9e6a-c19617cdf424

**Comment by dariokan82** [comment]
+1. Exact same issue. 

iOS 17.5.1, iPhone 14Pro

**Comment by mediaformat** [comment]
Fun fact leaving the article as soon as the spinner appears, and returning to it leaves the scroll position intact. Gotta be quick tho!

**Comment by bourvill** [comment]
Never reproduce this issue 😫

Can you share one or more entry with this bug ?

**Comment by simonv3** [comment]
Sure, 

It happened with this article https://dustycloud.org/blog/how-decentralized-is-bluesky/
as well as this one https://code.dccouncil.gov/us/dc/council/code/titles/29/chapters/8

**Comment by ronaldscott** [comment]
This happens every single time I use the app to read a longer article, with no exceptions. It is trivially reproducible through normal, regular use of the app. 

iOS latest at the time of this writing and iPhone 14 Pro Max.

**Comment by bourvill** [comment]
@ronaldscott No, it's not trivial, and it's not so easy to reproduce as you say. Every time I try to solve this problem, this problem does not occur.
Thank you for bringing constructive things to try to solve the problem and not simple banalities of your daily life

**Comment by simonv3** [comment]
@bourvill is there a way to see the logs in the app? Unfortunately I don't have an apple computer so I can't try building the app locally to check them.

**Comment by mediaformat** [comment]
Seems like some "lazy" process is triggering an app sync, I'm not sure what needs syncing in an in-progress article.

Anyhow, I agree that review local logs would help pinpoint the issue!

**Comment by bourvill** [comment]
Guy's! finally reproduced! working on fix!

**Comment by bourvill** [comment]
App is rolling out , test the latest version 7.5.0 and give me your feedback

**Comment by mediaformat** [comment]
This is still happening. 

Version 7.5.0 build 319

**Comment by mediaformat** [comment]
Just a theory, but my guess is the app tries to periodically sync the article scroll position with the server.

Perhaps the app should only GET the position on opening an article. And only POST the scroll position to the server.

I can't think of any other reason the app should load anything while reading an article.

**Comment by bourvill** [comment]
Position is not sync with the server, is only on local.

**Comment by foozzi** [comment]
Problem is still happening 
Version 7.6.0 build 319

iPhone 14 Pro Max, ios 18.4

**Comment by dadadajo** [comment]
Same issue here…

iPad 6th Generation 
iPadOS 17.7.6

Is there a way I can provide some logs?

**Comment by utrenkner** [comment]
@bourvill Thank you so much! I immediately tried it on several longer articles and the problem seems completely gone!!!

I hope everyone else has the same experience as me.

**Comment by mediaformat** [comment]
It seems to be working better than it was, however longer articles still tried to reload.
Here are the results of a few tests I did, with the estimated read time of the articles:
- [x] 12 mins 
- [ ] [19 mins](https://newsocialist.org.uk/transmissions/ai-the-new-aesthetics-of-fascism/)  
- [ ] [27 min](https://berjon.com/digital-sovereignty/)

**Comment by bourvill** [comment]
Can you share link of this articles ?

**Comment by mediaformat** [comment]
> Can you share link of this articles ?

They are linked above, but here they are in full:
- https://newsocialist.org.uk/transmissions/ai-the-new-aesthetics-of-fascism/
- https://berjon.com/digital-sovereignty/

**Comment by bourvill** [comment]
@mediaformat during this long reading, did you leave the application? Lock Screen? other thing than "just reading" ?

**Comment by mediaformat** [comment]
For the above tests, I scrolled through the articles quite quickly, not reading, but also not full speed.

Just now I was reading this one, normal speed, no interruptions. And twice it has refreshed before the 6th paragraph. 
https://matija.suklje.name/trying-out-koreader-and-wallabag-the-first-few-days-and-months

**Comment by bourvill** [comment]
Enter in TestFlight using this link 
https://testflight.apple.com/join/D4AGRSTX

In this beta you can find a new page home "Discover" is not finished at this time, but that includes a fix in article reader.

Give me a feedback,

Also during TestFlight you can test WallabagPlus for free.

**Comment by bourvill** [comment]
Thank your feedback @simmel, I have received your feedback, feedback give me only the screenshot and some basic information, like free space, carrier, version but no stack trace, stack trace occur only during crash.

I send another version in TestFlight channel, I remove some observation to the scroll position, the progress will not work in this version. like I said I'm blind, another big thank again to try to solve this problem

**Comment by simmel** [comment]
No refresh with 321 either. No progressbar though but not sure if that was one of the things you added back?

**Comment by bourvill** [comment]
In 321, I have restored the saving of the scroll position persisted. In 322 I will restore in ugly state the progress bar during reading.

**Comment by mediaformat** [comment]
322 is working well, have not been able to provoke a refresh.

**Comment by bourvill** [comment]
Really, thank you very much. We're making progress. There are still one or two more attempts left before we can better understand the problem.

The 323 will activate the progress bar smooth.

**Comment by mediaformat** [comment]
In 323 the loader issue comes back again. Curiously I fast scrolled an article (15m), and it didn't trigger. But when I read it at my normal speed it happened again.

**Comment by bourvill** [comment]
324 rollback the latest modifications, can you try ?

**Comment by mediaformat** [comment]
324 working much the same as 322

**Comment by simmel** [comment]
325 has no crash, restores position after going back to Entries and then returning to the article.

The blue "scrollbar"/positionbar doesn't change until after you stop scrolling. This also means that when returning to the article the position bar isn't visible at 0% until you scroll and stops and it is then correct and visible.

**Comment by proofrock** [comment]
Hi,
  Just in case you can use some more feedback. I installed wallabag yesterday, and the issue in the ios app was very evident. I could not read for more than two minutes. With v327 (that I just found in testflight) no issues so far; neither fast-scrolling for five minutes a long article, nor reading (at my pace) short and medium ones for twenty minutes. I will keep testing it, if it's useful for you. And thanks for your effort. 

    G. 

Ps: Maybe related, maybe an hallucination: when I used to have this issue, exiting Wallabag would slow down the ios web component to the point that I couldn't use it (regular Safari even). Killing wallabag would restore the functionality. I am aware that this is pretty incredible, and maybe it was due to something else, and just a coincidence. But this new version doesn't seem to cause this.

**Comment by simmel** [comment]
When I had the CPU issues I also noted that the application that was draining the battery was Safari and not wallabag but I’m guessing that due to the webview?

**Comment by proofrock** [comment]
> When I had the CPU issues I also noted that the application that was draining the battery was Safari and not wallabag but I’m guessing that due to the webview?

Makes sense... the strange thing is that wallabag was in background; but maybe the webview was active because of Safari. I am not an expert in iOS, but maybe there's a shared state of sorts.

**Comment by bourvill** [comment]
I will publish 327 to everyone.

Thank you all for your support and patience.

I will remove the new discovery home; it’s not finished and the bug is annoying, and many users give one star to the app on the App Store.

Did you find the new home discovery useful or interesting?

**Comment by simmel** [comment]
> I will remove the new discovery home; it’s not finished and the bug is annoying, and many users give one star to the app on the App Store.
> 
> Did you find the new home discovery useful or interesting?

I did try it before and my articles are such a mess that it’s not very helpful with my articles.

But yes it would be useful with something like that.

I think I would prefer two rows per category and sidescrolling if that’s possible.

Being able to filter and sort with your own criteria could be nice to (give me the 20 oldest articles that are over 20 min sorted by date so I can work on my backlog… XD)

**Comment by simmel** [comment]
328 doesn't show the scrolling bug nor high cpu usage.

Good job!
